### PR TITLE
feat(semver): add ignore merge flag

### PR DESCRIPTION
**Pull request by cakeinpanic**
Provided a solution to fix this issue
https://github.com/jscutlery/semver/issues/541

Set `ignoreMergeCommits` to true by default, but not sure if it should be set to `false` if it was flaky before :)

Anyway looks like it's a breaking change in some sense

### CODE REVIEW

**Comment by cakeinpanic** [comment]
@edbzn WDYT?

**Review by edbzn — CHANGES_REQUESTED** [review]
In my opinion, semver should ignore merge commits and more generally any commits that do not follow the semver spec but it is not necessary to add a new option to the executor for this, it is probably a breaking change but anyway, it should be the normal behavior.

File: packages/semver/src/executors/version/utils/git.ts [hunk 1]
```diff
@@ -9,41 +9,50 @@ import { logStep, _logStep } from './logger';
...
   since?: string;
 }): Observable<string[]> {
   return new Observable<string>((observer) => {
-    gitRawCommits({
+    const params: any = {
       from: since,
       format,
       path: projectRoot,
-    })
+      'full-history': true,
```
[the remark this answers is not in the document]
Comment (cakeinpanic) [reply packages/semver/src/executors/version/utils/git.ts:50]: I added explanation here https://github.com/conventional-changelog/conventional-changelog/issues/924

File: packages/semver/src/executors/version/schema.json [hunk 2]
```diff
@@ -103,6 +103,11 @@
         "description": "Type to ignore"
       }
     },
+    "ignoreMergeCommits": {
+      "description": "If to ignore merge commits on version bump calculation",
+      "type": "boolean",
+      "default": true
+    },
```
[the remark this answers is not in the document]
Comment (cakeinpanic) [reply packages/semver/src/executors/version/schema.json:110]: but what if someone adds significant information into merge commit titles?

File: packages/semver/src/executors/version/index.e2e.spec.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ import { getProjectDependencies } from './utils/get-project-dependencies';
 import { readPackageJson } from './utils/project';
 
 jest.mock('@nrwl/devkit');
-
+jest.setTimeout(1000000)
```
[the remark this answers is not in the document]
Comment (cakeinpanic) [reply packages/semver/src/executors/version/index.e2e.spec.ts:19]: it is, I just forgot to remove it after debugging in webstorm :)

**Comment by cakeinpanic** [comment]
@edbzn I'd be glad to ignore merge commits by default, but have some doubts about people probably relying on them...

File: packages/semver/src/executors/version/schema.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (cakeinpanic) [reply packages/semver/src/executors/version/schema.json]: I gave it a thought and decided to make this behavior built-in as you suggested

**Review by edbzn — APPROVED** [review]

**Pull request merged**